test/src/unit-capi-string_dims.cc

```diff
@@ -327,6 +328,7 @@ int StringDimsFx::array_create_wrapper(
           &array_schema) == TILEDB_OK);
 
   // Clean up.
+  tiledb_array_schema_free(&array_schema);
```

Comment:
While I don't think this would cause a problem (was it really being leaked?),
looks off-hand like it would be free'd after returning from the array_create_wrapper() call in both serialize_ cases 
(in the same 
```TEST_CASE_METHOD(
    StringDimsFx,
    "C API: Test sparse array with string dimensions, array schema",
    "[capi][sparse][string-dims][array-schema]") {
  SECTION("- No serialization") {
    serialize_ = false;
  }
  SECTION("- Serialization") {
    serialize_ = true;
  }
```
)

Reply:
The call to `tiledb_deserialize_array_schema` re-assigns `array_schema` to a newly-allocated address. This line frees the leak.

@bekadavis9 This is a valid fix, but it would be more readable if we used a new, auxiliary variable like we do on L300 (e.g. `tiledb_array_schema_t* new_array_schema2 = nullptr;`. Could you make this change for clarity?

Reply:
(ok, missed the '*' vs '**' difference [same name array_schema, different levels of indirection] between array_schema_load_wrapper()/array_create_wrapper() ).